modules/repository-gcs/src/internalClusterTest/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobStoreRepositoryTests.java

```diff
@@ -227,6 +228,7 @@ StorageOptions createStorageOptions(
                 ) {
                     StorageOptions options = super.createStorageOptions(gcsClientSettings, httpTransportOptions);
                     return options.toBuilder()
+                        .setStorageRetryStrategy(StorageRetryStrategy.getLegacyStorageRetryStrategy())
```

Comment:
Suggestion: 

We can extract `StorageRetryStrategy.getLegacyStorageRetryStrategy()` into a static method `storageRetryStrategy()` in `GoogleCloudStorageService`, so if happen to update the retry strategy in the future, we would need to do it only in one place.

Reply:
It might be worthwhile to wait until we find out whether the alternative to `LegacyStorageRetryStrategy` is one of the other two strategies available in the SDK, or a custom retry behaviour.

Reply:
Why do we have to set the strategy here? Shouldn't we already have set it in `super`? :)

Reply:
I thought the same! But the way we build the `StorageOptions` and then create a builder out of it again, seems to reset the strategy and we have to set it again. I tried a variation to just pass around the builder but that also ended up resetting to the `DefaultStorageRetryStrategy` (I think when we set the retry settings, it changes the strategy part too). Maybe just a funky non-intuitive API misunderstanding, but I thought it's probably ok as it is since it is only in two tests.

I'll give it another try, and if it still falls back to the `DefaultStorageRetryStrategy`, I'll merge it as is.